### DISCUSSION TITLE
Slow API responses once deployed to Vercel

### DISCUSSION THREAD

**Discussion by patjakubik** [vercel/vercel#5523] [Help] [answered]
I'm at the point where my app is finally ready to deploy to Vercel, however I noticed that API routes respond quite slow.

- On my local env: between 500 - 1000 ms
- Once deployed to Vercel: between 3000 - 4000 ms

Weirdly enough, other assets such as JavaScript load as fast as locally. Any ideas? I'm making a critical API call before the app is shown so it affects page load significantly.

**Comment by patjakubik** [comment, marked as answer]
Solved. Cause: lambda cold starts.

**Reply by styfle** [reply]
Consider setting `Cache-Control` headers if you expect the API response to remain the same for a period of time https://vercel.com/docs/edge-network/caching

When possible, use [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) (or your framework equivalent) to move the work from runtime to build time.

**Comment by vennson** [comment]
@patryk-smc 
make sure ur Function Region is the same to where your DB/API server is. (this is in the settings of ur project in vercel.)

i had 1-2s before, but now it's averaging 150ms (faster and more consistent than in my local dev)
my problem was my supabase region is singapore, but my vercel function region is US by default

<img width="1245" alt="Screenshot 2024-04-16 at 11 57 35 AM" src="https://github.com/vercel/vercel/assets/28529262/f69d1ca8-4040-43dc-ba80-a708240a109c">
